src/test/java/org/folio/dataimport/util/TryTest.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.dataimport.util;
+
+import io.vertx.core.Future;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.function.Supplier;
+
+@RunWith(JUnit4.class)
+public class TryTest {
+
+  @Test
+  public void shouldReturnSucceededFutureWhenTaskReturnedResult() {
+    Supplier<String> task = () -> "result string";
```

Comment:
What is the main reason to extract the task as a local variable?

- It's very short

- It is not express  the main Idea of the Try calss

- Extra line of code

Reply:
I played around it and rewrite it a little
```
public class TryTest {

  private static final String RESULT_STRING = "result string";

  @Test
  public void shouldReturnSucceededFutureWhenTaskReturnedResult() {
    Try.itGet(() -> RESULT_STRING)
      .setHandler(ar -> {
        Assert.assertTrue(ar.succeeded());
        Assert.assertEquals(RESULT_STRING, ar.result());
      });
  }

  @Test
  public void shouldReturnFailedFutureWhenTaskThrewException() {

    Try.itGet(this::throwException)
      .setHandler(ar -> {
        Assert.assertTrue(ar.failed());
        Assert.assertThat(ar.cause(), instanceOf(RuntimeException.class));
      });
  }

  @Test
  public void shouldReturnSucceededFutureWhenJobReturnedResult() {
    Try.itDo(future -> future.complete(RESULT_STRING))
      .setHandler(ar -> {
        Assert.assertTrue(ar.succeeded());
        Assert.assertEquals(RESULT_STRING, ar.result());
      });
  }

  @Test
  public void shouldReturnFailedFutureWhenJobThrewException() {

    Try.itDo(future -> throwException())
      .setHandler(ar -> {
        Assert.assertTrue(ar.failed());
        Assert.assertThat(ar.cause(), instanceOf(RuntimeException.class));
      });
  }

  private String throwException() {
    throw new RuntimeException();
  }
}
```